### PR TITLE
letsencrypt_directory, letsencrypt_lookup

### PR DESCRIPTION
**Pull request by cwells**
This PR addresses issue #66.  See comments for discussion.

### CODE REVIEW

**Comment by bastelfreak** [comment]
@cwells thanks for the PR. Can you check the used email address in your commits? It isn't associated with your github account.

File: lib/facter/letsencrypt_directory.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        cert = OpenSSL::X509::Certificate.new(File.new(pem).read)
+        san = cert.extensions.find { |e| e.oid == 'subjectAltName' }
+        names = san.value.split(',').map { |entry| entry.split(':')[1] }
+        names.each do |n|
+          certs[n] = path.to_s
+        end
+      end
+    end
+
+    certs
```
Comment (bastelfreak) [inline lib/facter/letsencrypt_directory.rb]: should we instead use a structured fact? Would make the function obsolete or easier

Comment (cwells) [reply lib/facter/letsencrypt_directory.rb]: Amending git history is a PITA. Unless there's some reason it matters, I'd rather just leave the email as-is. 

The fact is structured (it's a hash). While you can certainly index into it directly, the function also handles wildcards, which isn't something a structure is going to provide.

**Comment by thejambavan** [comment]
Please could this PR be merged? These facts would come in quite handy

**Comment by igalic** [comment]
I like this

the code looks fine (on a first glance, i haven't tested it yet) too
what's wrong with merging this and seeing what happens (before releasing)?

**Comment by cwells** [comment]
Closing due to inactivity.

**Pull request closed without merging**